src/test/java/com/google/cloud/spanner/pgadapter/JdbcMockServerTest.java

```diff
@@ -530,6 +583,37 @@ public void testDdl() throws SQLException {
     assertEquals(sql, updateDatabaseDdlRequests.get(0).getStatements(0));
   }
 
+  @Test
+  public void testDdlBatch() throws SQLException {
+    ImmutableList<String> statements =
+        ImmutableList.of(
+            "CREATE TABLE foo (id bigint primary key)",
+            "CREATE TABLE bar (id bigint primary key, value text)",
+            "CREATE INDEX idx_foo ON bar (text)");
+    addDdlResponseToSpannerAdmin();
+
+    try (Connection connection = DriverManager.getConnection(createUrl())) {
+      try (java.sql.Statement statement = connection.createStatement()) {
+        for (String sql : statements) {
+          statement.addBatch(sql);
+        }
+        int[] updateCounts = statement.executeBatch();
+        assertArrayEquals(new int[] {0, 0, 0}, updateCounts);
+      }
+    }
+
+    List<UpdateDatabaseDdlRequest> updateDatabaseDdlRequests =
+        mockDatabaseAdmin.getRequests().stream()
+            .filter(request -> request instanceof UpdateDatabaseDdlRequest)
+            .map(UpdateDatabaseDdlRequest.class::cast)
+            .collect(Collectors.toList());
+    assertEquals(1, updateDatabaseDdlRequests.size());
+    assertEquals(3, updateDatabaseDdlRequests.get(0).getStatementsCount());
```

Comment:
DDL is also automatically batched by PGAdapter when that is possible (in both simple and extended query mode)